diff --git a/build/configs/smb_multi.mk b/build/configs/smb_multi.mk
index 6524205..11eb7a1 100644
--- a/build/configs/smb_multi.mk
+++ b/build/configs/smb_multi.mk
@@ -59,6 +59,7 @@ SMB_PROFILE_LIST =   \
 
 # To enable Private MIBs, comment in the following line:
 # Custom/AddModules := private_mib private_mib_gen
+Custom/AddModules := demo
 
 # Custom/Stage2 := debug
 
diff --git a/vtss_appl/icfg/icfg_api.h b/vtss_appl/icfg/icfg_api.h
index ea0ba25..6ce14a9 100644
--- a/vtss_appl/icfg/icfg_api.h
+++ b/vtss_appl/icfg/icfg_api.h
@@ -488,8 +488,14 @@ typedef enum {
 #endif
 
     //------------------------- Interface Ethernet submode begin --------------
+#ifdef VTSS_SW_OPTION_DEMO
+    VTSS_ICFG_DEMO,
+#endif
+    
     VTSS_ICFG_INTERFACE_ETHERNET_BEGIN,
-
+#ifdef VTSS_SW_OPTION_DEMO
+    VTSS_ICFG_DEMO_INTERFACE_CONFIG,
+#endif
 #ifdef VTSS_SW_OPTION_VOICE_VLAN
     VTSS_ICFG_INTERFACE_ETHERNET_VOICE_VLAN,
 #endif
@@ -661,6 +667,10 @@ typedef enum {
     //------------------------- Interface VLAN submode begin ------------------
     VTSS_ICFG_INTERFACE_VLAN_BEGIN,
 
+#ifdef VTSS_SW_OPTION_DEMO
+    VTSS_ICFG_DEMO_INTERFACE_VLANCONFIG,
+#endif
+    
 #ifdef VTSS_SW_OPTION_IP
     VTSS_ICFG_IPV4_INTERFACE,
 #endif /* VTSS_SW_OPTION_IP */
diff --git a/vtss_appl/include/vtss/appl/module_id.h b/vtss_appl/include/vtss/appl/module_id.h
index e19ab31..9fe8636 100644
--- a/vtss_appl/include/vtss/appl/module_id.h
+++ b/vtss_appl/include/vtss/appl/module_id.h
@@ -203,6 +203,7 @@ enum {
     VTSS_MODULE_ID_DHCP6_RELAY          = 145,
     VTSS_MODULE_ID_DHCP6_SNOOPING       = 146,
     VTSS_MODULE_ID_IPV6_SOURCE_GUARD    = 147,
+    VTSS_MODULE_ID_DEMO    = 148,
 
     /*
      * INSERT NEW MODULE IDS HERE. AND ONLY HERE!!!
diff --git a/vtss_appl/main/main.cxx b/vtss_appl/main/main.cxx
index 1253933..5249129 100755
--- a/vtss_appl/main/main.cxx
+++ b/vtss_appl/main/main.cxx
@@ -470,6 +470,10 @@
 #include "vtss_optional_modules_api.h"
 #endif
 
+#if defined(VTSS_SW_OPTION_DEMO)
+#include "demo_api.h"
+#endif
+
 #if defined(CYGPKG_IO_FILEIO) && defined(CYGPKG_FS_RAM) && defined(VTSS_SW_OPTION_ICFG)
 #include <unistd.h>
 #include <cyg/fileio/fileio.h>
@@ -890,6 +894,12 @@ static struct {
     INITFUN(vtss_optional_modules_init)
 #endif
 
+#if defined(VTSS_SW_OPTION_DEMO)
+    INITFUN(demo_init)
+#endif
+
+
+    
 // **** NOTE: vtss_icfg_late_init must be last in this list ****
 #ifdef VTSS_SW_OPTION_ICFG
     INITFUN(vtss_icfg_late_init)
@@ -1296,6 +1306,11 @@ const char *error_txt(mesa_rc rc)
             txt = thread_load_monitor_error_txt(rc);
             break;
 #endif
+#ifdef VTSS_SW_OPTION_DEMO
+        case VTSS_MODULE_ID_DEMO:
+            txt = demo_error_txt(rc);
+            break;
+#endif
         default:
             txt = txt_default;
             break;
diff --git a/vtss_appl/util/vtss_module_id.cxx b/vtss_appl/util/vtss_module_id.cxx
index 269ee98..a25043f 100644
--- a/vtss_appl/util/vtss_module_id.cxx
+++ b/vtss_appl/util/vtss_module_id.cxx
@@ -181,6 +181,7 @@ const char * const vtss_module_names[VTSS_MODULE_ID_NONE + 1] =
     [VTSS_MODULE_ID_DHCP6_RELAY]      /* 145 */ = "DHCPv6Relay",
     [VTSS_MODULE_ID_DHCP6_SNOOPING]   /* 146 */ = "DHCPv6_Snooping",
     [VTSS_MODULE_ID_IPV6_SOURCE_GUARD] /*147 */ = "IPv6_Source_Guard",
+    [VTSS_MODULE_ID_DEMO] /*148 */ = "Demo",
 
     /* Add new module name above it. And please don't use space
        in module name, use underscore instead. */
@@ -551,6 +552,7 @@ const int vtss_priv_lvl_groups_filter[VTSS_MODULE_ID_NONE+1] =
     /*[VTSS_MODULE_ID_DHCP6_RELAY]      145 */ 1,
     /*[VTSS_MODULE_ID_DHCP6_SNOOPING]   146 */ 1,
     /*[VTSS_MODULE_ID_IPV6_SOURCE_GUARD] 147 */ 1,
+    /*[VTSS_MODULE_ID_DEMO] 148 */ 1,
 
     /* Hint:
      * For a new module, if the module needs an independent privilege level group
